convert2rhel/main.py

```diff
@@ -54,6 +54,10 @@ def main():
     utils.require_root()
 
     process_phase = ConversionPhase.INIT
+
+    # Archive the old log file before starting a new run
+    logger.archive_old_logger_files()
```

Comment:
It feels like at this level, we just want to call `initialize_logger()` and that would take care of all tasks related to setting up logging for us to run.  However, the present initialize_logger seems (appropriately) focused on setup that relates to the logging stdlib library.  Maybe we need a new function that sits in the middle?  What do you think of organizing it something like this:

``` python
# In main.py:
def main():
    logger.initialize_logger("convert2rhel.log")

# In logger.py:
def initialize_logger(log_name, log_dir=LOG_DIR):
    archive_old_logger_files(log_name)
    setup_logging_library(log_name, log_dir)

def setup_logging_library(log_name, log_dir=LOG_DIR):
    # Contents of the old initialize_logger
```